test/models/ocelot/stanzas/InputStanzaSpec.scala

```diff
@@ -61,6 +62,66 @@ class InputStanzaSpec extends BaseSpec {
     }
   }
 
+  "CurrencyInput " should {
+    "update the input label" in {
+
+      Input(expectedCurrencyStanza, Phrase("",""), None, None).get match {
+        case currencyInput: CurrencyInput =>
+          val labels = LabelCache()
+
+          val (nxt, updatedLabels) = currencyInput.eval("33", labels)
+          updatedLabels.updatedLabels(expectedCurrencyStanza.label).value shouldBe Some("33")
+        case _ => fail
+      }
+
+    }
+
+    "Determine invalid input to be incorrect" in {
+
+      Input(expectedCurrencyStanza, Phrase("",""), None, None).get match {
+        case currencyInput: CurrencyInput =>
+
+          currencyInput.validInput("a value") shouldBe None
+        case _ => fail
+      }
+
+    }
+
+    "Allow for coma separated 1000s" in {
+
+      Input(expectedCurrencyStanza, Phrase("",""), None, None).get match {
+        case currencyInput: CurrencyInput =>
+
+          currencyInput.validInput("123,345,768") shouldBe Some("123345768")
+        case _ => fail
+      }
+
+    }
+
+    "Allow -ve values" in {
+
+      Input(expectedCurrencyStanza, Phrase("",""), None, None).get match {
+        case currencyInput: CurrencyInput =>
+
+          currencyInput.validInput("-567,345") shouldBe Some("-567345")
+        case _ => fail
+      }
+
+    }
+
+    "Determine valid input to be correct" in {
+
+      Input(expectedCurrencyStanza, Phrase("",""), None, None).get match {
+        case currencyInput: CurrencyInput =>
+
+          currencyInput.validInput("33") shouldBe Some("33")
+        case _ => fail
+      }
+
+    }
+
+  }
+
```

Comment:
I would add a test for valid value with two decimal places (i.e. 1000.00) and invalid with three decimal places (i.e. 1000.000) for completeness (maybe one and zero decimal places too)